swarm/src/connection/pool.rs

```diff
@@ -237,6 +240,8 @@ where
         /// Addresses are dialed in parallel. Contains the addresses and errors
         /// of dial attempts that failed before the one successful dial.
         concurrent_dial_errors: Option<Vec<(Multiaddr, TransportError<TTrans::Error>)>>,
+        /// How long it took to establish this connection.
+        time_taken: std::time::Duration,
```

Comment:
```suggestion
        time_to_establish: std::time::Duration,
```

Would be more intuitive for me. //CC @thomaseizinger you always has good naming suggestions.

Reply:
In this case I think I disagree due to redundancy with context - this is a member of ConnectionEstablished. I think it's quite apparent what took that long to happen when we're describing a connection being established.

But of course I'll go along with whatever makes sense to others.

Reply:
> //CC @thomaseizinger you always has good naming suggestions.

A few names I can think of:

- `elapsed`
- `upgraded_in`
- `established_in`
- `since_accepted`
- `elapsed_since_accepted`

Reply:
@thomaseizinger  - The ones with "since" in them sound to me as if the stopwatch were still ticking, which it is not. That is, in my mind "since" means something like "up until now". This is the difference between a past time and a further-past time.

established_in sounds good enough to me, and it's pretty close to what @mxinden was hoping for too, so hopefully that's a good compromise.